src/middlewares/auth.ts

```diff
@@ -0,0 +1,63 @@
+import { NextFunction, Request, Response } from 'express';
+import { ForbiddenError, UnauthorizedError } from './error-handler.js';
+import jwt from 'jsonwebtoken';
+import memoize from 'memoizee';
+import axios from 'axios';
+import ms from 'ms';
+import * as config from '../config.js';
+import {egoUrl} from "../config.js";
+
+const getKey = memoize(
+	async (egoURL: string) => {
+		const response = await axios.get(egoURL);
```

Comment:
Looks like the `egoUrl` needs to point to the public key path... perhaps the ego URL should just be to the root of the API and then this function can add the path to the public key, since we may need the ego URL for other auth actions (token validation).

If these changes are part of a future PR for auth then lemme know and we'll ignore that for this PR.

Reply:
yes, all auth stuff is covered in the next PR.